src/main/java/seedu/address/ui/HomeCard.java

```diff
@@ -0,0 +1,237 @@
+    private static final String CONTACT_BOTTOM = "contacts on your FriendFolio";
+    private static final String MONEY_OWED_LABEL = "Money you owe";
+    private static final String OWED_MONEY_LABEL = "Money owed to you";
+    private static final String CHART_LABEL = "Financial Status";
+    private static final String AVAILABLE_LABEL = "Available Today";
+    private static final double CATEGORY_GAP = 120;
+    private static final double BAR_GAP = 0;
+    private static final double MOVE_DURATION = 200;
+    private static final double MOVE_INITIAL = 500;
+    private static final double MOVE_POP = 20;
+    private static final double FADE_DURATION = 300;
+    private static final double FADE_INITIAL = 0;
+    private static final double FADE_ULTIMATE = 1;
+    private ObservableList<Person> personList;
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
+
+    @FXML
+    private HBox cardPane;
+    @FXML
+    private Label time;
+    @FXML
+    private Label second;
+    @FXML
+    private Label date;
+    @FXML
+    private Label contactTop;
+    @FXML
+    private Label contactAmount;
+    @FXML
+    private Label contactBottom;
+    @FXML
+    private Label chartLabel;
+    @FXML
+    private Label availableToday;
+    @FXML
+    private BarChart<String, Number> chart;
+    @FXML
+    private ListView<Person> availableList;
+    private TranslateTransition moveTransition;
+    private TranslateTransition moveBackTransition;
+    private FadeTransition fadeInTransition;
+
+    /**
+     * Creates a {@code HomeCard} with the given Person List.
+     */
+    public HomeCard(ObservableList<Person> personList) {
+        super(FXML);
+        this.personList = personList;
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm");
+        DateTimeFormatter secondFormatter = DateTimeFormatter.ofPattern(":ss");
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("EEEE, MMMM d");
+
+        time.setText(LocalDateTime.now().format(timeFormatter));
+        second.setText(LocalDateTime.now().format(secondFormatter));
+        date.setText(LocalDateTime.now().format(dateFormatter));
+
+        Timeline timeline = new Timeline(new KeyFrame(Duration.seconds(1), event -> {
+            LocalDateTime currentTime = LocalDateTime.now();
+            time.setText(currentTime.format(timeFormatter));
+            second.setText(currentTime.format(secondFormatter));
+            date.setText(currentTime.format(dateFormatter));
+        }));
+        timeline.setCycleCount(Timeline.INDEFINITE);
+        timeline.play();
```

Comment:
Should this be extracted out as a function or something? I notice you have `setUpList` and `setUpAnimation` below and this doesn't really seem like single level of abstraction.

Reply:
You can have one for time, one for contacts, one for the bar chart, one for the available, then one to set up animations.